### ISSUE TITLE
Adding support for "git+ssh://git@github.com/expressjs/express.git#semver:^4.16.0" format

### ISSUE THREAD

**Issue by sunknudsen** [raineorshine/npm-check-updates#820]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 10.17`

---------------------------

## Steps to Reproduce

According to npm [docs](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#git-urls-as-dependencies), `#semver:^4.16.0"` (or `#semver:^v4.16.0"`) format is used to specify semver when using git URLs as dependencies.

According to some tests (see [npm-check-updates-test](https://github.com/sunknudsen/npm-check-updates-test) repo for steps to reproduce), `npm-check-updates` doesn't support this format.

## Current Behavior

```json
{
  "dependencies": {
    "express": "git+ssh://git@github.com/expressjs/express.git#semver:^4.16.0"
  }
}
```

```console
$ npm ls express
npm-check-updates-npm-check-updates-test@1.0.0 /Users/sunknudsen/Desktop/npm-check-updates-test
└── express@4.16.0  (git+ssh://git@github.com/expressjs/express.git#f974d22c66d3cd91634ddaba1ef8bcaa8e49daf4)

$ ncu
Checking /Users/sunknudsen/Desktop/npm-check-updates-test/package.json
[====================] 1/1 100%

All dependencies match the latest package versions :)
```

## Expected Behavior

```json
{
  "dependencies": {
    "express": "git+ssh://git@github.com/expressjs/express.git#semver:^4.16.0"
  }
}
```

```console
$ npm ls express
npm-check-updates-npm-check-updates-test@1.0.0 /Users/sunknudsen/Desktop/npm-check-updates-test
└── express@4.16.0  (git+ssh://git@github.com/expressjs/express.git#f974d22c66d3cd91634ddaba1ef8bcaa8e49daf4)

$ ncu
Checking /Users/sunknudsen/Desktop/npm-check-updates-test/package.json
[====================] 1/1 100%

 express  ^4.16.0  →  ^4.17.1     

Run ncu -u to upgrade package.json
```

**Update:** forgot to mention how thankful I am for this project. 🤗

**Comment by raineorshine** [comment]
Thanks for reporting! I don’t have my computer at the moment, but my first guess is that it’s a false negative from [parse-github-url](https://github.com/jonschlinkert/parse-github-url) which is used [here](https://github.com/raineorshine/npm-check-updates/blob/9ad29937ae6924b33cc3da5b5ce9f5be5bca7ed8/lib/package-managers/gitTags.js#L11) to build the url for git remote tags. 

You could investigate and confirm this or I’m happy to in a couple days when I get my computer back.

**Comment by raineorshine** [comment]
Fixed and published in `v11.1.4`

**Comment by sunknudsen** [comment]
This is amazing! Thanks @raineorshine. Just tested `11.1.4` and works as expected.

Impressed by how caring you are for the `npm-check-updates` user base. ❤️
